File: 2020/2/python/part1.py
import re
from os import sys

with open('../input.txt', 'r') as f:
    passwords = f.readlines()

count = 0
for password in passwords:
    data = password.split(' ')

    try:
        min = int(data[0].split('-')[0])
        max = int(data[0].split('-')[1])
    except:
        continue

    letter = data[1][0]

    cur = data[2].strip()

    occur = len([m.start() for m in re.finditer(letter, cur)])


    if min <= occur <= max:
        count+=1

print(count)

File: 2020/2/cpp/part1.cpp
#include <vector>
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

int main() {

    vector<string> vec;

    // 1-3 l: llll

    string pass, s;
    int min, max, occ, res;
    char junk, letter;
    while (cin) {
        cin >> min >> junk >> max >> letter >> junk >> pass;

        occ = count(pass.begin(), pass.end(), letter);

        if (min <= occ && occ <= max) {
            res++;
        }
    }

    std::cout << res << std::endl;

    return 0;
}
